server/src/main/java/org/elasticsearch/search/profile/query/QueryProfiler.java

```diff
@@ -28,20 +27,20 @@
 public final class QueryProfiler extends AbstractProfiler<QueryProfileBreakdown, Query> {
 
     /**
-     * The root Collector used in the search
+     * The root CollectorResult used in the search
      */
-    private Supplier<CollectorResult> collectorResultSupplier;
+    private CollectorResult collectorResult;
```

Comment:
This supplier was really only needed because it was difficult to keep track of the collector (or manager) that provides us with the top level collector result tree. As that is now part of the result returned by the search method, it is very simple to set the result after search has been called which allows us to remove the need for the supplier.